homeassistant/components/iskra/entity.py

```diff
@@ -1,63 +1,59 @@
 """Base entity for Iskra devices."""
 
 from homeassistant.helpers.device_registry import DeviceInfo
-from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import DOMAIN, MANUFACTURER
+from .coordinator import IskraDataUpdateCoordinator
 
 
-class IskraDevice(Entity):
+class IskraEntity(CoordinatorEntity):
     """Representation a base Iskra device."""
 
     _attr_should_poll = True
+    _attr_has_entity_name = True
 
-    def __init__(self, device, gateway, config_entry):
+    def __init__(self, coordinator: IskraDataUpdateCoordinator) -> None:
         """Initialize the Iskra device."""
+        super().__init__(coordinator)
+        device = coordinator.device
         self._state = None
-        self._is_available = True
         self._serial = device.serial
         self._model = device.model
         self._fw_version = device.fw_version
         self._device_name = f"{self._serial}"
-        self._remove_unavailability_tracker = None
         self._device = device
-        self.gateway = gateway
-
         self._is_gateway = self._device.is_gateway
+        self._gateway = device.parent_device
         self._device_id = self._serial
 
     @property
-    def device_id(self):
+    def device_id(self) -> str:
         """Return the device id of the Iskra device."""
         return self._device_id
```

Comment:
This one can be removed if you store the variable in `self.device_id` instead

Reply:
done